### ISSUE TITLE
Searching messages by receival date uses wrong parameter

### ISSUE THREAD

**Issue by Zombaya** [microsoftgraph/microsoft-graph-docs#2873]
# Article
[Use query parameters to customize responses](https://developer.microsoft.com/en-us/graph/docs/concepts/query_parameters#search-parameter)

# Issue
The documentation states that you can search for messages by using KQL and the field `receivedDateTime`.  This however does not work.  If you replace this field by `received`, it does work as it should.

> You can specify the following properties on a message in a $search criterion:
>
> ...
> receivedDateTime
> ...

# Examples
The examples should return the message with subject `Hello world!`
## Working query using `received`
```
https://graph.microsoft.com/v1.0/me/messages?$search="received>2017-09-13T21:40:00Z AND received<2017-09-13T21:41:00Z"
```
[example](https://developer.microsoft.com/en-us/graph/graph-explorer?request=me/messages?$search="received>2017-09-13T21:40:00Z%20AND%20received<2017-09-13T21:41:00Z"&method=GET&version=v1.0&GraphUrl=https://graph.microsoft.com)
## Failing query using `receivedDateTime`
```
https://graph.microsoft.com/v1.0/me/messages?$search="receivedDateTime>2017-09-13T21:40:00Z AND receivedDateTime<2017-09-13T21:41:00Z"
```
[example](https://developer.microsoft.com/en-us/graph/graph-explorer?request=me/messages?$search="receivedDateTime>2017-09-13T21:40:00Z%20AND%20receivedDateTime<2017-09-13T21:41:00Z"&method=GET&version=v1.0&GraphUrl=https://graph.microsoft.com)

**Comment by Zombaya** [comment]
I don't know whether the documentation or the API should be changed.  

I should note that I found the lack of an example also did not help and I believe it should be very helpful if an example for searching by receival date would be added to this part of the documentation and to the examples of the Graph Explorer.

**Comment by angelgolfer-ms** [comment]
@Zombaya , I made some clarification in the docs (https://developer.microsoft.com/en-us/graph/docs/concepts/query_parameters#search-parameter). It may take up to a day for the update to refresh on the live site. Thanks for bringing up the issue.
I can add an example next week as well.

**Comment by Zombaya** [comment]
@angelgolfer-ms, Thanks for clearing this up.

However, I noticed you also changed the property `hasAttachments`  to `has`.  However, I do believe `hasAttachments:true` works as intended.

## Query using has:true
[https://graph.microsoft.com/v1.0/me/messages?$search="has:true"&$select=subject,hasAttachments&$top=5](https://developer.microsoft.com/en-us/graph/graph-explorer?request=me/messages?$search="has:true"&$select=subject,hasAttachments&$top=5&method=GET&version=v1.0&GraphUrl=https://graph.microsoft.com)
```json
{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users('48d31887-5fad-4d73-a9f5-3c356e68a038')/messages(subject,hasAttachments)",
    "value": [
        {
            "@odata.etag": "W/\"CQAAAA==\"",
            "id": "AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAAAAAAEMAAAiIsqMbYjsT5e-T7KzowPTAAAYbsdXAAA=",
            "hasAttachments": true,
            "subject": "Northwind Traders Proposal"
        }
    ]
}
```
## Query using has:false
[https://graph.microsoft.com/v1.0/me/messages?$search="has:false"&$select=subject,hasAttachments&$top=5](https://developer.microsoft.com/en-us/graph/graph-explorer?request=me/messages?$search="has:false"&$select=subject,hasAttachments&$top=5&method=GET&version=v1.0&GraphUrl=https://graph.microsoft.com)
```json
{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users('48d31887-5fad-4d73-a9f5-3c356e68a038')/messages(subject,hasAttachments)",
    "value": []
}
```
## Query using hasAttachments:true
[https://graph.microsoft.com/v1.0/me/messages?$search="hasAttachments:true"&$select=subject,hasAttachments](https://developer.microsoft.com/en-us/graph/graph-explorer?request=me/messages?$search="hasAttachments:true"&$select=subject,hasAttachments&method=GET&version=v1.0&GraphUrl=https://graph.microsoft.com)
```json
{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users('48d31887-5fad-4d73-a9f5-3c356e68a038')/messages(subject,hasAttachments)",
    "value": [
        {
            "@odata.etag": "W/\"CQAAAA==\"",
            "id": "AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAAAAAAEMAAAiIsqMbYjsT5e-T7KzowPTAAC6dGcJAAA=",
            "hasAttachments": true,
            "subject": "Your Office 365 billing statement"
        },
        {
            "@odata.etag": "W/\"CQAAAA==\"",
            "id": "AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAAAAAAEMAAAiIsqMbYjsT5e-T7KzowPTAAAYbsdWAAA=",
            "hasAttachments": true,
            "subject": "Please Forward Contoso patent document"
        },
        {
            "@odata.etag": "W/\"CQAAAA==\"",
            "id": "AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAAAAAAEMAAAiIsqMbYjsT5e-T7KzowPTAAAYbsdNAAA=",
            "hasAttachments": true,
            "subject": "Northwind Proposal"
        },
        {
            "@odata.etag": "W/\"CQAAAA==\"",
            "id": "AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAAAYbtbpAAAiIsqMbYjsT5e-T7KzowPTAAAYbv4XAAA=",
            "hasAttachments": true,
            "subject": "Approval Request"
        }
    ]
}
```
## Query using hasAttachments:false
[https://graph.microsoft.com/v1.0/me/messages?$search="hasAttachments:false"&$select=subject,hasAttachments&$top=5](https://developer.microsoft.com/en-us/graph/graph-explorer?request=me/messages?$search="hasAttachments:false"&$select=subject,hasAttachments&$top=5&method=GET&version=v1.0&GraphUrl=https://graph.microsoft.com)
```json
{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users('48d31887-5fad-4d73-a9f5-3c356e68a038')/messages(subject,hasAttachments)",
    "@odata.nextLink": "https://graph.microsoft.com/v1.0/me/messages?$search=%22hasAttachments%3afalse%22&$select=subject%2chasAttachments&$top=5&$skiptoken=MSZZVlF4YVU1cVVtaFpha3B0VG1rd05WcHFXVE5NVkZFd1RsZFZkRmxVUlhoYVF6QjRUV3BSTVZsVVVUVk9WRVpwVFdwTmJXTjZNREU9",
    "value": [
        {
            "@odata.etag": "W/\"CQAAAA==\"",
            "id": "AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAAAAAAEMAAAiIsqMbYjsT5e-T7KzowPTAADo_fh3AAA=",
            "hasAttachments": false,
            "subject": "You have late tasks!"
        },
        {
            "@odata.etag": "W/\"CQAAAA==\"",
            "id": "AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAAAAAAEMAAAiIsqMbYjsT5e-T7KzowPTAADop-3GAAA=",
            "hasAttachments": false,
            "subject": "You have late tasks!"
        },
        {
            "@odata.etag": "W/\"CQAAAA==\"",
            "id": "AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAAAAAAEMAAAiIsqMbYjsT5e-T7KzowPTAADj5wuzAAA=",
            "hasAttachments": false,
            "subject": "You have late tasks!"
        },
        {
            "@odata.etag": "W/\"CQAAAA==\"",
            "id": "AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAADHHBFaAAAiIsqMbYjsT5e-T7KzowPTAADj5w-KAAA=",
            "hasAttachments": false,
            "subject": null
        },
        {
            "@odata.etag": "W/\"CQAAAA==\"",
            "id": "AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAAAAAAEMAAAiIsqMbYjsT5e-T7KzowPTAADj5wuyAAA=",
            "hasAttachments": false,
            "subject": "You have tasks due today!"
        }
    ]
}
```

**Comment by angelgolfer-ms** [comment]
Hi @Zombaya , I included hasAttachments and added examples (PR #2919 and #2928 ). Hope this will help. As usual, it may take a day or so to have the updates displayed on the live site. Thanks for your support!

**Comment by Zombaya** [comment]
Great work!  I believe those examples will be of great help for people implementing a search-functionality.
